### DISCUSSION TITLE
interfacing OV7670 with ESP32 through I2C

### DISCUSSION THREAD

**Discussion by Mnsh1368** [micropython/micropython#15438] [Hardware & Peripherals]
![Uploading ۲۰۲۴۰۷۰۹_۱۸۵۶۳۷.j
![Uploading ۲۰۲۴۰۷۰۹_۱۸۵۷۳۴.jpg…]()
pg…]()
Thank you very much in advance for your help.
I am trying to connect OV7670 with ESP32 Wroom V1 using I2C, but when I run my related Micropython code, [19] ENODEV will appear. 
I didn't find out what the problem is and how solving it.
Can anyone help me with this issue?

**Comment by shariltumin** [comment]
Can you send me the picture of your OV7670  module? Generic ESP32 firmware does not support OV7670, you need special firmware for that. Beside this, you need ESP32 board that support PSRAM in order to effectively buffer the image data produced by the OV7670.

**Reply by Mnsh1368** [reply]
Hi,
Thanks a lot for your answer.
The pictures have been attached following, 
I used the camera without FIFO, and the ESP32 version is Wroom32 v1.
What is the special firmware for interfacing with OV7670? The firmware I used was "ESP32_GENERIC-20240222-v1.22.2". 
![۲۰۲۴۰۷۰۹_۱۸۵۷۳۴](https://github.com/user-attachments/assets/e75c4c6b-016a-4b5d-adef-73cd69192fcf)
![۲۰۲۴۰۷۲۳_۲۰۲۶۵۶](https://github.com/user-attachments/assets/550f903a-b8e1-458e-98b4-39eaa2e4d5cc)
![۲۰۲۴۰۷۲۳_۲۰۲۷۳۵](https://github.com/user-attachments/assets/c92c75c3-c08b-4b50-ba65-ea491b57001a)

**Reply by shariltumin** [reply]
ESP32 Wroom V1 board has no PSRAM. You need an ESP32 WROVER-E board. The firmware with camera driver needs PSRAM. You cannot use standard I2C to connect to the OV7670. As you can see, the OV7670 has signal, control and data pins. To connect to the OV7670 and get images from it, you need a costume made MicroPython firmware. Currently, the firmware in this repository only supports the OV2640.

**Reply by Mnsh1368** [reply]
Hi,
Thanks for your useful information.
Since your last answer, I have searched and investigated many boards and cameras on the Internet, but the more I search, the more frustrated. 
My project is to build a robot that can do object detection tasks using data received from the camera. First, I decided to use ESP32 Dev kit v1 and OV7670 for the task. Regarding your answers, I found the components are not true options for this project because not developing the specific firmware.

what is your idea about choosing the right components for this task?
